src/pages/api/user.ts

```diff
@@ -0,0 +1,178 @@
+                to: email,
+                from: 'cop43318@gmail.com',
+                template_id: 'd-32c150b4de8043edba973cd21ace99f5',
+                // Information being sent in the email.
+                dynamic_template_data: {
+                    firstName: firstName,
+                    ahjst: emailTok
+                }
+            }
+            sgMail
+                .send(msg)
+                .then(() => {
+                    console.log('Email sent')
+                })
+                .catch((error: string) => {
+                    console.error(error)
+                })
+
+        // Hash Password.
+        const hashedPass = hashPass(password);
+
+        // Save New User.
+        const userData = {name: firstName, lastName, email, password:hashedPass, verifyEmail:false, emailToken: emailTok, resPassword:0, resPassToken:''};
+        const savedUser = await prisma.user.create({data:userData});
+        return res.status(200).json({error: ""});
+    } 
+    
+    // Updates user information.
+    if (req.method === "PUT") {
+        // Checks JWT token.
+        // const token = await getToken({req});
+        // if (!token)
+        //     return res.status(401).json({ error: "User is not logged in." });
+
+        const { firstName, lastName, email } = req.body;
+
+        // Error: Email was not received.
+        if(!email) 
+            return res.status(400).json({ error: "Email was not recieved." });
+
+        // Error: Not all fields are filled out.
+        if(!firstName || !lastName)
+            return res.status(400).json({ error: "Please fill out all fields." });
+
+        // Error: Name values are not strings.
+        if((typeof firstName !== 'string') || (typeof lastName !== 'string'))
+            return res.status(400).json({ error: "Please input a proper name." });
+
+        // If all checks are passed.
+
+        const user = await prisma.user.update({
+            where: {
+                email,
+            },
+            data: {
+                name: firstName,
+                lastName,
+            }
+        })
+        if(user)
+            return res.status(200).json({error: ""});
+        else
+            return res.status(200).json({error: "Could not update information"})
+    }
+
+    // Deletes a user.
+    if (req.method === "DELETE") {
+        // Checks JWT token.
+        // const token = await getToken({req});
+        // if (!token)
+        //     return res.status(401).json({ error: "User is not logged in." });
+        
+        const { userID } = req.body;
+
+        // Error: UserID is not valid.
+        if (!userID || typeof userID != "string")
+            return res.status(401).json({ error: "UserID is not a string." });
+
+        // Deletes a user.
+        const deleteUser = await prisma.user.delete({
+            where: {
+                id: userID,

```

Comment:
token.id and everything else can go. Unless you are doing this for the admin. In such case you need to add more logic here. `if (token.role === "ADMIN") then delete body.id OTHERWISE delete token.id on the user`.